compute_endpoint/tests/utils.py

```diff
@@ -100,9 +101,50 @@ def kill_manager():
     import os
     import signal
 
+    import psutil
+
     manager_pid = os.getppid()
+    my_pid = os.getpid()
+
+    # We need to kill the manager and it's children manually
+    # until we have better cleanup on the parsl side: sc_28400
+    current_process = psutil.Process(manager_pid)
+
+    children = current_process.children(recursive=True)
     os.kill(manager_pid, signal.SIGKILL)
+    for child in children:
+        if child.pid != my_pid:
+            os.kill(child.pid, signal.SIGKILL)
+
+    os.kill(my_pid, signal.SIGKILL)
```

Comment:
Consider killing the process group instead.  The kernel will do the heavy lifting and no need for faffing with `psutil` or avoiding your own pid until the end.  A single call:
```python
import os
import signal

manager_pg = os.getpgid(os.getppid())
os.killpg(manager_pg, signal.SIGKILL)
```

Reply:
I did try this, and it didn't work. I do not dig into why. For now, I think this is a stopgap until we get the changes @rjmello put into parsl's manager that does better worker cleanup on getting SIGINT.

Reply:
Fixed. Using the PGID of the manager to kill the PG works better than the current approach.